lib/virtio_mmio/virtio_mmio.c

```diff
@@ -391,10 +480,25 @@ struct virtqueue *virtio_mmio_setup_virtqueue(struct virtio_device *vdev,
 		metal_assert(maxq != 0);
 		metal_assert(maxq >= vq->vq_nentries);
 		virtio_mmio_write32(vdev, VIRTIO_MMIO_QUEUE_NUM, vq->vq_nentries);
+#if defined(HVL_VIRTIO)
+		if (vmdev && (vmdev->hvl_mode == 1)) {
+			virtio_mmio_hvl_ipi();
+		}
+#endif
```

Comment:
Do you need to call virtio_mmio_hvl_ipi twice times? 
This one seems useless by looking at the code below

Reply:
Fixed.